src/components/modal/confirm-edit-connection-string.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Actions from 'actions';
+import { Modal } from 'react-bootstrap';
+import { TextButton } from 'hadron-react-buttons';
+
+import styles from '../connect.less';
+
+/**
+ * Question text.
+ */
+const QUESTION = 'Are you sure you want to edit your connection string?';
+
+/**
+ * The note.
+ */
+const NOTE = 'Editing this connection string will reveal your credentials.';
+
+/**
+ * Confirm edit a connection string modal.
+ */
+class ConfirmEditConnectionString extends PureComponent {
+  static displayName = 'ConfirmNewPipelineComponent';
+
+  static propTypes = {
+    isEditURIConfirm: PropTypes.bool.isRequired
+  };
+
+  /**
+   * Handles clicks on the `Confirm` button.
+   */
+  onConfirm = () => {
+    Actions.onEditURIConfirmed();
+  };
+
+  /**
+   * Closes the current modal.
+   */
+  onClose = () => {
+    Actions.onEditURICanceled();
+  };
+
+  /**
+   * Render the component.
+   *
+   * @returns {React.Component} The component.
+   */
+  render() {
+    return (
+      <Modal show={this.props.isEditURIConfirm}>
+        <Modal.Header closeButton>
```

Comment:
I think we need to add a handler for the close button. An `onHide={this. onClose}` on the <Modal> should do the trick.

Reply:
Hehe, nice catch!

Reply:
Fixed!